BitFaster.Caching/Scoped.cs

```diff
@@ -45,7 +45,7 @@ public bool TryCreateLifetime(out Lifetime<T> lifetime)
                 var oldRefCount = this.refCount;
 
                 // If old ref count is 0, the scoped object has been disposed and there was a race.
-                if (this.isDisposed || oldRefCount.Count == 0)
+                if (IsDisposed || oldRefCount.Count == 0)
```

Comment:
I think there is another potential race as follows:

1. Thread A: check IsDisposed returns false
2. Thread B: Dispose, IsDisposed now true
3. Thread A: IncrementCount, return lifetime containing disposed value

I am not able to provoke this case with the soak test.